MIPs/mip-3.md

```diff
@@ -0,0 +1,184 @@
+
+- **Undefined Parameters**: Parameters like `from`, `to`, `userAddress`, etc. are essential for the `wallet_swapAsset` method to work correctly. If not provided, the `validateParams` function would throw an error saying `"${property} property of ${name} is not defined"`. (This validation will change on the future when we allow multiple swap tokens, since the architecture of this rpc method allows it). See more details about error codes in the api-spec: https://github.com/MetaMask/api-specs/pull/201
+
+  ```markdown
+  validateParams(from[0], [tokenAddress], 'from');
+  validateParams(to, [tokenAddress], 'to');
+  ```
+
+- **Invalid User Address**: If a non-existent address is provided in `userAddress`, then the method will throw an error 'This address does not exist'. See more details about error codes in the api-spec: https://github.com/MetaMask/api-specs/pull/201
+
+  ```markdown
+  if (!dappConnectedAccount) {
+  throw rpcErrors.invalidParams('This address does not exist');
+  }
+  ```
+
+- **Unsupported Multiple Tokens Swap**: As of now, it supports only a single token swap. In case of multiple tokens provided in the `from` object, an 'Currently we do not support multiple tokens swap' error is thrown. See more details about error codes in the api-spec: https://github.com/MetaMask/api-specs/pull/201
+
+  ```markdown
+  if (from.length > 1) {
+  throw rpcErrors.invalidParams(
+  'Currently we do not support multiple tokens swap',
+  );
+  }
+  ```
+
+- **Inactive or Unsupported Swaps**: If the swap is inactive or is not possible on the current chain, an alert with the message 'Swap is not active or is not possible on this chain' will be triggered. See more details about error codes in the api-spec: https://github.com/MetaMask/api-specs/pull/201
+
+  ```markdown
+  const isSwappable = isSwapsAllowed(chainId) && swapsIsLive;
+  if (!isSwappable) {
+  Alert.alert(`Swap is not available on this chain ${networkName}`);
+  throw rpcErrors.methodNotSupported(
+  `Swap is not available on this chain ${networkName}`,
+  );
+  }
+  ```
+
+5. Security: Dapp Developers should ensure that the token addresses provided in the method call are secure and valid for the current chain ID. They should also inform users about the potential risks and considerations when performing token swaps.
+   On MetaMask platforms we have warnings in place that say to the user to be carefull when a token is not that trustworthy. We check if the number of occurrences is more than one, If not we show an alert the we obligate the user to ready and press a button to be able to swap.
```

Comment:
```suggestion
5. Security: In MetaMas we have warnings in place to alert the user when a token is not trustworthy and add an additional confirmation step. Despite this, dApp developers should still ensure that the token addresses provided in the method call are secure and valid for the current chain ID. They should also inform users about the potential risks and considerations when performing token swaps.
```

maybe?

Reply:
Sounds good!

Reply:
Addressed on this [commit](https://github.com/MetaMask/metamask-improvement-proposals/pull/25/commits/d64432f9dc185bbd80153fa1950aa18ee8b74db7)